backend/main/mutations/UserMutations.py

```diff
@@ -0,0 +1,55 @@
+import graphene
+
+from main.types.UserType import UserType
+from django.contrib.auth import get_user_model
+    
+class CreateUser(graphene.Mutation):
+    class Arguments:
+        username = graphene.String(required=True)
+        email = graphene.String(required=True)
+
+    user = graphene.Field(UserType)
+
+    def mutate(self, info, username, email):
+        user = get_user_model()(username=username, email=email)
+        user.save()
+        return CreateUser(user=user)
+
+
+class UpdateUser(graphene.Mutation):
+    class Arguments:
+        id = graphene.ID(required=True)
+        username = graphene.String(required=True)
+        email = graphene.String(required=True)
+
+    user = graphene.Field(UserType)
+
+    def mutate(self, info, id, username=None, email=None):
+        try:
+            user = get_user_model().objects.get(pk=id)
+        except get_user_model().DoesNotExist:
+            raise Exception("user not found")
+
+        if username is not None:
+            user.username = username
+        if email is not None:
+            user.email = email
+
+        user.save()
+        return UpdateUser(user=user)
+
+
+class DeleteUser(graphene.Mutation):
+    class Arguments:
+        id = graphene.ID(required=True)
+
+    success = graphene.Boolean()
+
+    def mutate(self, info, id):
```

Comment:
All mutations here are pretty powerful, and currently anyone with access to the API can perform them as often as they want. I know that this is just for testing/learning purposes and that we will delete this before it goes into production, but I would feel a bit safer if we only allowed these mutations to be performed by superusers.

```suggestion
@classmethod
def mutate(cls, root: None, info: ResolveInfo, id: str):
    current_user: User | AnonymousUser | None  = info.context.user
    if not current_user.is_superuser:
        return DeleteUser(success=false, errors=["User is not authorised to delete other users."])
```